### PR TITLE
LPS-195135 Convert old AUI to new js in bookmarks-web

### PR DESCRIPTION
**Pull request by DiegoHu97**
[Link to ticket](https://liferay.atlassian.net/browse/LPS-195135)

## Steps

- Go to Content & Data > Bookmarks
- Create a folder and a bookmark
- See the gif
ezgif com-video-to-gif

### CODE REVIEW

**Comment by liferay-echo** [comment]
Thank you for submitting this PR. 👏

Make sure it follows the [Echo Pull Request Template](https://gist.github.com/liferay-echo/5a4455285ff1800f9da146816d85e67d#file-pull-request-template-md) if you haven't already done so. 

Learn more about it [here](https://liferay.atlassian.net/wiki/spaces/ENGECHO/pages/2027455014/Echo+Pull+Request+Etiquette).

**Comment by DiegoHu97** [comment]
ci:test:sf

File: modules/apps/bookmarks/bookmarks-web/src/main/resources/META-INF/resources/bookmarks/js/BookmarksDragAndDrop.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+	function moveToTrash() {
+		processAction('move_to_trash', editEntryURL);
+	}
+
+	function moveToFolder(object) {
+		const dropTarget = object.targetItem;
+
+		const selectedItems = object.selectedItems;
+
+		const folderId = dropTarget.getAttribute('data-folder-id');
```
Comment (victorg1991) [inline modules/apps/bookmarks/bookmarks-web/src/main/resources/META-INF/resources/bookmarks/js/BookmarksDragAndDrop.js:29]: For accessing data attributes it is preferable to use `dropTarget.dataset.folderId` :)

File: modules/apps/bookmarks/bookmarks-web/src/main/resources/META-INF/resources/bookmarks/js/BookmarksDragAndDrop.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+	searchContainer.registerAction('move-to-trash', moveToTrash);
+
+	function moveToTrash() {
+		processAction('move_to_trash', editEntryURL);
+	}
+
+	function moveToFolder(object) {
+		const dropTarget = object.targetItem;
+
+		const selectedItems = object.selectedItems;
```
Comment (victorg1991) [inline modules/apps/bookmarks/bookmarks-web/src/main/resources/META-INF/resources/bookmarks/js/BookmarksDragAndDrop.js:27]: We can use deestructuring here like:

```javascript
const {dropTarget, selectedItems} = object;
```

File: modules/apps/bookmarks/bookmarks-web/src/main/resources/META-INF/resources/bookmarks/js/BookmarksDragAndDrop.js [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+				processAction('move', moveEntryURL);
+			}
+		}
+	}
+
+	function processAction(action, url, redirectUrl) {
+		const form = document.getElementById(`${namespace}fm`);
+
+		redirectUrl = redirectUrl || location.href;
```
Comment (victorg1991) [inline modules/apps/bookmarks/bookmarks-web/src/main/resources/META-INF/resources/bookmarks/js/BookmarksDragAndDrop.js:50]: RedirectUrl is not passed in any of the processAction invocations so we can remove this line and the parameter

File: modules/apps/bookmarks/bookmarks-web/src/main/resources/META-INF/resources/bookmarks/js/BookmarksDragAndDrop.js [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+		if (form.querySelector(`#${namespace}javax-portlet-action`)) {
+			form.querySelector(
+				`#${namespace}javax-portlet-action`
+			).value = action;
+		}
+		else {
+			form.querySelector(`#${namespace}cmd`).value = action;
+		}
+
+		form.querySelector(`#${namespace}redirect`).value = redirectUrl;
```
Comment (victorg1991) [inline modules/apps/bookmarks/bookmarks-web/src/main/resources/META-INF/resources/bookmarks/js/BookmarksDragAndDrop.js:63]: It seems that  javax-portlet-action never exist in the DOM, so we can remove that check. Also for setting values in the form, there is an utility available `setFormValues` that you can use and you'd avoid having to set the namespace yourself :)

**Comment by victorg1991** [comment]
@DiegoHu97 a couple of suggestion to better follow liferay patterns :)

**Comment by DiegoHu97** [comment]
@victorg1991 Updated! Thanks 😄

**Comment by DiegoHu97** [comment]
ci:test:sf

**Comment by victorg1991** [comment]
Nice job @DiegoHu97 !

**Comment by ealonso** [comment]
ci:forward:force

**Comment by ealonso** [comment]
ci:forward:force

**Pull request closed without merging**